elasticsearch-evolution-core/src/main/java/com/senacor/elasticsearch/evolution/core/internal/migration/execution/MigrationServiceImpl.java

```diff
@@ -185,9 +185,20 @@ List<ParsedMigrationScript> getPendingScriptsToBeExecuted(Collection<ParsedMigra
             ParsedMigrationScript parsedMigrationScript = orderedScripts.get(i);
             if (!protocol.getVersion().equals(parsedMigrationScript.getFileNameInfo().getVersion())) {
                 throw new MigrationException(String.format(
-                        "The logged execution in the Elasticsearch-Evolution history index at position %s is version %s and in the same position in the given migration scripts is version %s! Out of order execution is not supported. Or maybe you have added new migration scripts in between or have to cleanup the Elasticsearch-Evolution history index manually",
+                        "The logged execution in the Elasticsearch-Evolution history index at position %s " +
+                                "is version %s and in the same position in the given migration scripts is version %s! " +
+                                "Out of order execution is not supported. Or maybe you have added new migration scripts " +
+                                "in between or have to cleanup the Elasticsearch-Evolution history index manually",
                         i, protocol.getVersion(), parsedMigrationScript.getFileNameInfo().getVersion()));
             }
+            // failed scripts can be edited and retried, but successfully executed scripts may not be modified afterwards
+            if (protocol.isSuccess() && protocol.getChecksum() != parsedMigrationScript.getChecksum()) {
```

Comment:
You should add a configuration in `ElasticsearchEvolutionConfig` to enable/disable this check and pass it to MigrationServiceImpl. You can enable this by default, but this have to be documented in the README.md at first in the `## 6 changelog` section for the next `SNAPSHOT` (like the corresponding config section `## 5 Configuration options`)